okhttp-tests/src/test/java/com/squareup/okhttp/internal/http/URLConnectionTest.java

```diff
@@ -425,6 +425,11 @@ private void testServerClosesOutput(SocketPolicy socketPolicy) throws Exception
     connection1.setReadTimeout(100);
     assertContent("This connection won't pool properly", connection1);
     assertEquals(0, server.takeRequest().getSequenceNumber());
+
+    // Give the server time to enact the socket policy if it's one that could happen after the
```

Comment:
See comment above. It probably does try to recover, but the tests are no typically written to actually deal with the different points when the connection may be closed, only the common case when it happens before the 2nd request has been issued.